src/applications/edu-benefits/complaint-tool/actions/schoolSearch.js

```diff
@@ -16,6 +19,36 @@ export function clearSearch() {
   };
 }
 
+export function restoreFromPrefill({ institutionQuery, institutionSelected, page, searchInputValue }) {
+  return dispatch => {
+    dispatch({
+      type: RESTORE_FROM_PREFILL_STARTED,
+      institutionQuery,
+      institutionSelected,
+      page,
+      searchInputValue
+    });
+
+    fetchInstitutions({ institutionQuery, page }).then(({ error, payload }) => {
```

Comment:
Can you give me an example of what you mean? 

My understanding is that `fetchInstitutions` calls `apiRequest` which takes an error callback argument and calls that with its own `catch`. I think that since `apiRequest` has an outer most `catch` in its promise chain, no exceptions / failed promises would escape it. `fetchInstitutions`'s error callback just returns the error on an { error } object. 

I definitely need to add a catch here, regardless.

Reply:
Perhaps something like this.

`fetchInstitutions({ institutionQuery, page, onDone, onError })`

```
export function fetchInstitutions({ institutionQuery, page, onDone, onError }) {
  const fetchUrl = appendQuery('/gi/institutions/search', {
    name: institutionQuery,
    include_address: true, // eslint-disable-line camelcase
    page
  });

  return apiRequest(
    fetchUrl,
    null,
    payload => onDone(payload),
    error => onError(error)
  );
}
```

Reply:
Thanks I think that's much cleaner. updated.